src/mqtt/qmqtt_router.h

```diff
@@ -47,11 +47,14 @@ class Q_MQTT_EXPORT Router : public QObject
     Q_OBJECT
 public:
     explicit Router(Client *parent = 0);
+    ~Router();
 
     RouteSubscription *subscribe(const QString &route);
+    void unsubscribe();
 
 private:
     Client *_client;
+    QString _route;
```

Comment:
this should contain all routes passed to ubscribe(const QString &route) during the Router's life

Reply:
I agree. 

If you think about this a little longer, should there also be an automatic unsubscribe if one of the created `RouteSubscription` objects is deleted?